gcp_variant_transforms/libs/beam_util.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Wrapper for Beam classes, e.g., Metrics.
+
+This is used to avoid direct dependency on Beam in our library methods/classes
+such that they are more modular and easier to test. Preferably instances
+of classes in this module should be injected into library classes/methods
+instead of direct instantiation.
+"""
+
+from __future__ import absolute_import
+
+import logging
+
+from apache_beam.runners import runner  # pylint: disable=unused-import
+from apache_beam.metrics import Metrics
+from apache_beam.metrics.metric import MetricsFilter
+
+_METRICS_NAME_SPACE = 'VT_metrics_name_space'
+
+
+class BaseCounter(object):
+  """The interface of counter objects which does nothing.
+
+  This also acts as a nop implementation.
+  """
+
+  def inc(self, n=1):
+    # type: (int) -> None
+    """Subclass implementations should do increment by `n`."""
+    pass
+
+
+class _CounterWrapper(BaseCounter):
+  """A wrapper for Beam counters."""
+
+  def __init__(self, counter_name):
+    # type: (str) -> None
+    self._counter_name = counter_name
+    self._counter = Metrics.counter(_METRICS_NAME_SPACE, counter_name)
+
+  def inc(self, n=1):
+    # type: (int) -> None
+    """Increments the counter by `n`"""
+    self._counter.inc(n)
+
+  def get_counter(self, pipeline_result):
+    # type: (runner.PipelineResult) -> int
+    """Extracts and returns the value of this counter.
+
+    Note that this goes through all counters every time it is called to find the
+    matching counter (see `query` implementation). So if the goal is to just
```

Comment:
What use case do you envision for get_counter()? How is it used in PR#126?

Reply:
You are right that it is not used in this PR (and not even PR #126) except in unit-tests. But I originally started by logging the counters one by one and that is useful in those cases. Since it is also unit-tested and the logic is not trivial (it took me some time to find what is the right way of getting a single counter value), I decided to keep it but if you feel strongly about it, I can remove it either.

Reply:
Yes, I think it's a good practice to remove unused code, otherwise we will end up with unused code, no one remember about it. Can we put what you found valuable in the comment instead?

Reply:
Done.
For now I just removed it.